### PR TITLE
[FIX] partner_phone_search: Return None when name is False/empty

### PR DESCRIPTION
**Pull request by igallart**
When we do not have a name, it returns None, and another module that read the result could fail.
For example, in the previous commit of this module: res + = super (ResPartner, self) .name_search ..., will fail.

### CODE REVIEW

**Review by tarteo — COMMENTED** [review]
I think it's better to do the super before ```if name``` and than return the result at the end. And change the result of super in ```if name``` something like ```result = res + result```.

**Comment by igallart** [comment]
> I think it's better to do the super before `if name` and than return the result at the end. And change the result of super in `if name` something like `result = res + result`.

If you does the super before name and name is not false, you has doing two super unnecessarily. And, the first super is not valid in 

> args += [('id', 'not in', partners.ids)]
> return res + super(ResPartner, self).name_search(
> name, args=args, operator=operator, limit=limit_rest)

because args are differents.

**Review by tarteo — APPROVED** [review]
@igallart Ah yes, makes sense. Thanks!

**Comment by rafaelbn** [comment]
Hello @igallart , is it posiible to increase  codecov/patch — 66.66% of diff hit (target 97.69%) ? :smile:

**Comment by igallart** [comment]
> Hello @igallart , is it posiible to increase codecov/patch — 66.66% of diff hit (target 97.69%) ? smile

@rafaelbn, I'm God! Mmmm, I don't know how is that, I only repair the error of module...

File: partner_phone_search/models/res_partner.py [hunk 1]
```diff
@@ -15,14 +15,15 @@ def name_search(self, name, args=None, operator='ilike', limit=100):
...
             res = partners.name_get()
             if limit:
                 limit_rest = limit - len(partners)
             else:
                 limit_rest = limit
             if limit_rest or not limit:
                 args += [('id', 'not in', partners.ids)]
-                res += super(ResPartner, self).name_search(
-                    name, args=args, operator=operator, limit=limit_rest)
+                res += super().name_search(name, args, operator, limit_rest)
```
Comment (NL66278) [inline partner_phone_search/models/res_partner.py]: You should call keyword arguments with keywords. So limit your change to removing the unneeded arguments from the super() call.

File: partner_phone_search/models/res_partner.py [hunk 2]
```diff
@@ -15,14 +15,15 @@ def name_search(self, name, args=None, operator='ilike', limit=100):
...
             return res
+        else:
+            return super().name_search(name, args, operator, limit)
```
Comment (NL66278) [inline partner_phone_search/models/res_partner.py]: Same here,

**Review by NL66278 — CHANGES_REQUESTED** [review]
See comments: use keywords for arguments that are defined as keyword arguments.

**Review by NL66278 — CHANGES_REQUESTED** [review]
The code itself looks mostly OK, but please add a test where you search on phone or mobile. Also test the situation that phone/mobile/email are not present.

To increase performance, you could check for digits in the name argument, before searching on phone/mobile. You could also check for '@' in the name argument before searching on email. Actually searching on phone/mobile and on email are mutually exclusive considering this.

In summary: if no digits or @ in name, just return super. If @ in name search on email. Else search on phone or mobile. Note that an email address can contain digits, but a phone number cannot contain a @.

**Comment by igallart** [comment]
> The code itself looks mostly OK, but please add a test where you search on phone or mobile. Also test the situation that phone/mobile/email are not present.
> 
> To increase performance, you could check for digits in the name argument, before searching on phone/mobile. You could also check for '@' in the name argument before searching on email. Actually searching on phone/mobile and on email are mutually exclusive considering this.
> 
> In summary: if no digits or @ in name, just return super. If @ in name search on email. Else search on phone or mobile. Note that an email address can contain digits, but a phone number cannot contain a @.

@NL66278 , thanks by your comments, but I don't have time to implement all those improvements. I did the pull request to solve a fatal error under certain circumstances, therefore, if someone can make the proposed improvements it would be perfect, but I can't dedicate that time to a module that I don't think is essential either.
